src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3250 @@
+                    if (runAsBehavior == DSCResourceRunAsCredential.Mandatory)
+                    {
+                        if (string.Equals(prop.Name, "PsDscRunAsCredential", StringComparison.OrdinalIgnoreCase))
+                        {
+                            keyProp.Mandatory = true;
+                        }
+                    }
+
+                    if (valueMap != null && keyProp.Values.Count > 0)
+                    {
+                        if (valueMap.Count != keyProp.Values.Count)
+                        {
+                            s_tracer.WriteLine(
+                                "DSC CreateDynamicKeywordFromClass: the count of values for qualifier 'Values' and 'ValueMap' doesn't match. count of 'Values': {0}, count of 'ValueMap': {1}. Skip the keyword '{2}'.",
+                                keyProp.Values.Count, valueMap.Count, keyword.Keyword);
+                            return null;
+                        }
+
+                        for (int index = 0; index < valueMap.Count; index++)
+                        {
+                            string key = keyProp.Values[index];
+                            string value = valueMap[index];
+
+                            if (keyProp.ValueMap.ContainsKey(key))
+                            {
+                                s_tracer.WriteLine(
+                                    "DSC CreateDynamicKeywordFromClass: same string value '{0}' appears more than once in qualifier 'Values'. Skip the keyword '{1}'.",
+                                    key, keyword.Keyword);
+                                return null;
+                            }
+
+                            keyProp.ValueMap.Add(key, value);
+                        }
+                    }
+
+                    keyword.Properties.Add(prop.Name, keyProp);
+                }
+            }
+
+            // update specific keyword with range constraints
+            UpdateKnownRestriction(keyword);
+
+            return keyword;
+        }
+
+        /// <summary>
+        /// Update range restriction for meta configuration keywords
+        /// the restrictions are for
+        /// ConfigurationModeFrequency: 15-44640
+        /// RefreshFrequency: 30-44640.
+        /// </summary>
+        /// <param name="keyword"></param>
+        private static void UpdateKnownRestriction(DynamicKeyword keyword)
+        {
+            if (
+                string.Equals(keyword.ResourceName, "MSFT_DSCMetaConfigurationV2",
+                    StringComparison.OrdinalIgnoreCase)
+                ||
+                string.Equals(keyword.ResourceName, "MSFT_DSCMetaConfiguration",
+                    StringComparison.OrdinalIgnoreCase))
+            {
+                if (keyword.Properties["RefreshFrequencyMins"] != null)
+                {
+                    keyword.Properties["RefreshFrequencyMins"].Range = new Tuple<int, int>(30, 44640);
```

Comment:
Would be better to have these values be const

Reply:
Updated.